### ISSUE TITLE
FancyBox3 video in hidden <Div> not playing in Opera and Safari (but other browsers OK)

### ISSUE THREAD

**Issue by MicJarek** [fancyapps/fancybox#1857]
Hi,

I've upgraded to FancyBox3 and it's great for the most part. Images display flawlessly, and to display video I use a hidden `<Div>`. The video displays fine in all the latest browser versions I tested (Firefox, IE, and Edge), but NOT in Opera or Safari. In both of these browsers the sound plays, but the video just hangs with a black screen. Note: I'm also using the latest jquery-3.2.1.min.js so hopefully that's not an issue.

Here is a page to see what I mean (http://www.beatonmyheart.com/myheart/Procedure.aspx) 
And code I used is pretty basic:

    <a data-fancybox="gallery" data-src="#video1" data-width="480" data-height="384" data-caption="Caption goes here..." >
        <img src="data/video/Video1 - Thumb.png" align="left" alt="" ></a>
        <div hidden id="video1" style="margin: 1px; padding: 1px; background-color: #000000;">
              <video width="480" height="384" controls ><source src="data/video/Video1.mp4" type="video/mp4">Your browser does not support HTML5 video.</video>
        </div>

Has anyone else tried to display an .mp4 video via FancyBox3 using a hidden `<Div>` and is having the same problem? Any feedback/suggestions would be greatly appreciated.

Tks... :)

**Comment by fancyapps** [comment]
Hi,

I think there is something with your video files, because other videos play without any problem.

**Comment by MicJarek** [comment]
Hi FancyApps,

Thanks for the feedback, but if the video plays fine in Firefox, IE, and Edge (on PC/laptop), then doesn't that imply that it's not the video itself, but rather the browser(s) that might be causing this?

Take for example the single video on the page: [http://www.beatonmyheart.com/myheart/Diagnosis5.aspx](url). I used MediaInfo utility (see image below) to get the details of the .mp4 video, and it appears fine. 

So, do you have any suggestions on how to correct the video, if you're confident it's the video that is the problem? Or better yet, do you have a sample video that you can point me to, so I can test and verify that it works in ALL the browsers, and does NOT display the green tint on mobile devices (since you closed the other issue #1858.)

Tks again for your help...
![image](https://user-images.githubusercontent.com/17038597/36697232-548d4722-1afb-11e8-8b3f-8c8725b16d03.png)

**Comment by kloneets** [comment]
@MicJarek try to open this video on safari and opera without fancybox (through html5 video api, of course). If it plays well, then fancybox is doing something wrong. 
In your case video is coded with AVC, but mostly browsers support only H.264 or HEVC. So - you need to re-code your video for browser compatibility.

**Comment by fancyapps** [comment]
The script does what it does, the rest is up to you and your browser.

**Comment by MicJarek** [comment]
Thanks @kloneets and @fancyapps for your input. I'm still learning the differences between the different video formats/compressions, but after some googling and playing around, I got the video to work everywhere, except on the PC/Laptop Opera browser (I read that it doesn't natively support the H.264). But the Opera mobile works fine, and the green tint is also now gone!!

In case anyone else is curious and has similar problems, here is what happened. I tried converting the original .flv files (as flash is being phased out) into .mp4 using the open source VLC player/converter. The converter allowed me to select the H.264 codec, but when it exported, it was what the MediaInfo screen capture above shows.

Instead of VLC, I ended up using an online video conversion utility [https://www.onlineconverter.com/flv-to-mp4](url) and the result was what you see in the screen capture below. Notice that it is still "Format/Info : Advanced Video Codec", but at the bottom, it says "Writing library : x264 core 152 r19 ba24899", and I'm guessing that's what was needed because the videos now play fine.

Thanks again for your help.
Cheers... :)

![image](https://user-images.githubusercontent.com/17038597/36752029-b478ba84-1bb6-11e8-935e-bcdee68e2489.png)
